BranchSDK/src/BranchIO/DeviceInfo.cpp

```diff
@@ -342,31 +356,61 @@ std::string DeviceInfo::osDisplayName()
     return "Unknown";
 }
 
-
-std::string DeviceInfo::osVersion()
+bool DeviceInfo::getOSVersionInfo(OSVERSIONINFOEX& ver)
 {
-    std::string osVersion;
-    std::ostringstream osVersionStream;
-    
-    OSVERSIONINFOEX ver;
     ver.dwOSVersionInfoSize = sizeof(ver);
 #pragma warning(suppress : 4996)
     if (GetVersionEx((OSVERSIONINFO*)&ver) == 0)
     {
         DWORD error = GetLastError();
         BRANCH_LOG_D("GetVersionEx failed with error: " << error);
+        return false;
     }
-    else
+    return true;
+}
+
+std::string DeviceInfo::osVersion()
+{
+    OSVERSIONINFOEX ver;
+    if (!getOSVersionInfo(ver))
     {
-        osVersionStream << ver.dwMajorVersion << "." << ver.dwMinorVersion << " (Build " << (ver.dwBuildNumber & 0xFFFF);
-        wstring ws(ver.szCSDVersion);
-        string csdVersion(StringUtils::wstring_to_utf8(ws));
-        if (!csdVersion.empty()) osVersionStream << ": " << csdVersion;
-        osVersionStream << ")";
+        return "Unknown";
     }
-   
+
+    std::ostringstream osVersionStream;
+    osVersionStream << ver.dwMajorVersion << "." << ver.dwMinorVersion;
     return osVersionStream.str();
 }
 
+std::string DeviceInfo::osBuildNumber()
+{
+    OSVERSIONINFOEX ver;
+    if (!getOSVersionInfo(ver))
+    {
+        return "Unknown";
+    }
+
+    std::ostringstream buildVersionStream;
+    buildVersionStream << (ver.dwBuildNumber & 0xFFFF);
+    return buildVersionStream.str();
+}
+
+std::string DeviceInfo::osPlatformVersion()
+{
+        int highestVersion = 0;
+        const int MAX_VERSION = 30;  // Max Version - Will fail for Versions after 30. Highest available Major version right now is 19
```

Comment:
Can you point to the API documentation where this is defined? In internal follow up ticket.